File: solutions/fsharp/leap/2/Leap.fs
module Leap

// on every year that is evenly divisible by 4
//   except every year that is evenly divisible by 100
//     unless the year is also evenly divisible by 400

let leapYear (year: int) : bool =
    let divBy4 = year % 4 = 0
    let divBy100 = year % 100 = 0
    let divBy400 = year % 400 = 0

    divBy4 && (not divBy100 || (divBy100 && divBy400))

File: solutions/ocaml/leap/1/leap.ml
let leap_year y = 
    let divisible n = y mod n = 0 in
    let div_by_400 = divisible 400 in
    let div_by_100 = divisible 100 in
    let div_by_4 = divisible 4 in
    div_by_400 || (div_by_4 && not div_by_100)
